core/templates/domain/summary/exploration-summary-backend-api.service.ts

```diff
@@ -17,77 +17,91 @@
  * from the backend.
  */
 
-require('services/alerts.service.ts');
-require('services/validators.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ExplorationSummaryBackendApiService', [
-  '$http', '$q', 'AlertsService',
-  'ValidatorsService', 'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, AlertsService,
-      ValidatorsService, EXPLORATION_SUMMARY_DATA_URL_TEMPLATE) {
-    var _fetchExpSummaries = function(
-        explorationIds, includePrivateExplorations, successCallback,
-        errorCallback) {
-      if (!explorationIds.every(ValidatorsService.isValidExplorationId)) {
-        AlertsService.addWarning('Please enter a valid exploration ID.');
+import { AlertsService } from 'services/alerts.service';
+import { ValidatorsService } from 'services/validators.service';
+import { AppConstants } from 'app.constants';
 
-        var returnValue = [];
-        for (var i = 0; i < explorationIds.length; i++) {
-          returnValue.push(null);
-        }
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationSummaryBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private alertsService: AlertsService,
+    private validatorsService: ValidatorsService) {}
 
-        if (errorCallback) {
-          errorCallback(returnValue);
-        }
-        return;
+  _fetchExpSummaries(
+      explorationIds: string[],
+      includePrivateExplorations: boolean,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: string[]) => void): void {
+    if (!explorationIds.every(expId =>
+      this.validatorsService.isValidExplorationId(expId, true))) {
+      this.alertsService.addWarning('Please enter a valid exploration ID.');
+      var returnValue = [];
+      for (var i = 0; i < explorationIds.length; i++) {
+        returnValue.push(null);
       }
-
-      var explorationSummaryDataUrl = EXPLORATION_SUMMARY_DATA_URL_TEMPLATE;
-
-      $http.get(explorationSummaryDataUrl, {
-        params: {
-          stringified_exp_ids: JSON.stringify(explorationIds),
-          include_private_explorations: JSON.stringify(
-            includePrivateExplorations)
-        }
-      }).then(function(response) {
-        var summaries = angular.copy(response.data.summaries);
+      if (errorCallback) {
+        errorCallback(returnValue);
+      }
+      return;
+    }
+    var explorationSummaryDataUrl =
+    AppConstants.EXPLORATION_SUMMARY_DATA_URL_TEMPLATE;
+    this.httpClient.get(explorationSummaryDataUrl, {
+      params: {
+        stringified_exp_ids: JSON.stringify(explorationIds),
+        include_private_explorations: JSON.stringify(
+          includePrivateExplorations)
+      }
+    }).toPromise().then((response: Object[]) => {
+      var summaries = angular.copy(response);
```

Comment:
Don't use the `angular` keyword in any migrated code. You don't need to copy the response in this case.
```suggestion
    }).toPromise().then((summaries: Object[]) => {
```

Reply:
done!